hazelcast/src/main/java/com/hazelcast/flakeidgen/impl/FlakeIdGeneratorProxy.java

```diff
@@ -91,7 +82,12 @@
         this.logger = nodeEngine.getLogger(getClass());
 
         FlakeIdGeneratorConfig config = nodeEngine.getConfig().findFlakeIdGeneratorConfig(getName());
-        epochStart = EPOCH_START - (config.getIdOffset() >> (BITS_SEQUENCE + BITS_NODE_ID));
+        bitsTimestamp = config.getBitsTimestamp();
+        bitsSequence = config.getBitsSequence();
+        bitsNodeId = config.getBitsNodeId();
```

Comment:
We should maybe throw here if these three add to more than 64, some of the timestamp bits will be pushed out.